### PR TITLE
feat: Adding java controller example using spring-boot

### PR DESCRIPTION
**Pull request by salaboy**
Adding Java / Spring Boot controller example. 

This project can be built using maven (which comes with the example) and optionally deployed with Knative Functions, but normal Kubernetes deployment can be used as well. 

The example reconciles a new CRD which checks some websites to be up, and if they are all up a new Deployment is created. 

The project also includes a test and a README explaining how to use this example with MetaController. 

PS: feedback is appreciated

### CODE REVIEW

File: examples/java/spring-boot/internet-controller/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Internet Controller (MetaController) function
+
+This project was created to demonstrate how to use MetaController and Knative Functions. This controller shows how to monitor the internet and trigger some internet production tests only when the application is healthy.
```
Comment (grzesuav) [inline examples/java/spring-boot/internet-controller/README.md:3]: `how to monitor the internet` -> please be more specific, what exactly is being  monitored ? Some upstream service etc ?

File: examples/java/spring-boot/internet-controller/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+For this example, the way to check if the internet is healthy is by sending HTTP requests to different websites, instead of relying on Kubernetes Resources. 
+
+To create the Deployment that will run the internet ready production tests, we use MetaController children definitions. 
+
+To run this project you need: 
+- `helm` version 3.8+ installed 
+- MetaController installed
+  - to install with helm you can run the following commands
+  - `HELM_EXPERIMENTAL_OCI=1 helm pull oci://ghcr.io/metacontroller/metacontroller-helm --version=v2.2.5` to fetch the metacontroller chart
```
Comment (grzesuav) [inline examples/java/spring-boot/internet-controller/README.md:16]: `v2.2.5` is very old, please switch to the lates release

File: examples/java/spring-boot/internet-controller/config/controller-deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+apiVersion: v1
+kind: namespace
+metadata:
+  name: metacontroller
```
Comment (grzesuav) [inline examples/java/spring-boot/internet-controller/config/controller-deploy.yaml:4]: I think it will be created via helm chart anyway

**Review by grzesuav — CHANGES_REQUESTED** [review]
HI, several comments :
* please remove `DS_Store` files
* please add a `test.sh` like https://github.com/metacontroller/metacontroller/blob/master/examples/globalconfigmap/test.sh which will install, execute and clean up controller

**Comment by grzesuav** [comment]
@AmitKumarDas @mikesmithgh WDYT ? My only concern is  that do not demonstrates any particular behavior of metacontroller

**Comment by AmitKumarDas** [comment]
> @AmitKumarDas @mikesmithgh WDYT ? My only concern is that do not demonstrates any particular behavior of metacontroller

I am a bit hesitant on adding codebases into this main repo esp. since the same has to be maintained e.g. security vulnerabilities, dependency updates & stuff. I understand that this is part of examples but the amount of files involved here is significant.

Again I understand sprint boot is quite popular at end users' & above comment is certainly not meant to discourage the community.

**Comment by salaboy** [comment]
I will fix the requested changes, and I can make the example a bit more simple.

The current example shows how to create children resources and how to reconcile based on the payload that is created by metacontroller. I am happy to show more metacontroller features.. can you please suggest some? 

I am also happy to help maintaining this. 

My intention by creating the PR was to get feedback because it will be great to have some java specific examples here to make it easy for the java folks to get involved.

**Comment by AmitKumarDas** [comment]
> I will fix the requested changes, and I can make the example a bit more simple.
> 
> The current example shows how to create children resources and how to reconcile based on the payload that is created by metacontroller. I am happy to show more metacontroller features.. can you please suggest some?
> 
> I am also happy to help maintaining this.
> 
> My intention by creating the PR was to get feedback because it will be great to have some java specific examples here to make it easy for the java folks to get involved.

@salaboy It will be great if you can maintain the example.

**Comment by salaboy** [comment]
@AmitKumarDas ok.. I am happy to maintain or find someone else to help me maintain the example.. If you ever need changes to be applied I am also happy to be assigned to issues related with the example too.

**Comment by salaboy** [comment]
I am currently off until monday.. so I will work on the changes as soon as I get back to work

**Comment by salaboy** [comment]
Crazy week here.. I am still catching up.. but I haven't forgotten this one.. I will work on this asap

**Comment by grzesuav** [comment]
closing for now @salaboy

**Pull request closed without merging**